### ISSUE TITLE
Please clarify wether HMR should work in the new version

### ISSUE THREAD

**Issue by swernerx** [faceyspacey/extract-css-chunks-webpack-plugin#72]
I just want to figure out wether HMR should work with the newest version. I am using the newest version of both this plugin and webpack. It seems that CSS changes are correctly triggering HMR, but in the end nothing happens. This is the client-side log output:

```
｢hot｣ App updated. Recompiling src/components/card/Card.module.css
log.js:48 ｢hot｣ webpack: Compiling (<unnamed compiler>)
log.js:48 ｢hot｣ App Updated, Reloading Modules
log.js:48 ｢hot｣ Checking for updates to the bundle.
log.js:48 ｢hot｣ No Modules Updated.
log.js:48 ｢hot｣ App is up to date.
```

Re-loading the app manually works fine though.

This is my loader config for CSS files:

```js
{
  test: /\.module\.(css|sss|pcss)$/,
  use: [
    extractCssLoader,
    "cache-loader",
    "thread-loader",
    {
      loader: "css-loader",
      options: {
        import: false,
        minimize: false,
        sourceMap: true
      }
    },
    {
      loader: "postcss-loader",
      options:{
        sourceMap: true
      }
    }
  ]
}
```

Any hints? Previously with `mini-css-extract-plugin` I used the `style-loader` during development. Is this also the way to go with this plugin?

**Comment by zackljackson** [comment]
Yes the app does does reload,can you send me your full webpack config? Also are you using webpack 4?

**Comment by swernerx** [comment]
I mean the idea with HMR is that it should not reload, right?

My Webpack config is somewhat generated. 

My issue is that with the old approach of using "style-loader" in development HMR works as expected while when using this plugin it just doesn't seem to correctly trigger the required changes.

**Comment by swernerx** [comment]
Seems like Webpack itself is doing fine:

```
98% after emittingCompleted client bundle in 2.914s!
ℹ ｢hot｣: webpack: Compiling Done
ℹ ｢wdm｣: Version: webpack 4.11.0
Time: 13914ms
Built at: 05.06.2018 16:37:12
Entrypoint main = main-c29587bba029864b102d.css index.js main.ef57d98abc586b5abd48.hot-update.js
ℹ ｢wdm｣: Compiled successfully.
```

There seems to be a CSS file being generated. Looks like an issue with the client side.

**Comment by zackljackson** [comment]
Ahh yeah I get it. Using something like webpack merge. So the main thing to enable hot, is that there’s some kind of ref in the entry point. 

Let me get on my computer and I’ll send you examples. If I had a consoles logged version of your webpack it would help. But I’ll send you what I have locally. 

Alternatively I can just add hot:true to the plugin

**Comment by zackljackson** [comment]
Are you trying to hash the css files? can’t hash them in development mode

**Comment by swernerx** [comment]
Yep, it's Webpack v4.11.0 right now. Everything using latest versions.

**Comment by zackljackson** [comment]
It would be helpful to see the entry config and what you pass to entries

**Comment by swernerx** [comment]
I do hash them I figure.

**Comment by zackljackson** [comment]
I’m on my phone at the moment. But class check the build and see if there’s any console warnings that were thrown

**Comment by swernerx** [comment]
Hashing was clearly an issue, conceptually, but does not fix the issue when disabled.

New output:

```
ℹ ｢hot｣: webpack: Bundle Invalidated
ℹ ｢wdm｣: Compiling...
Building client bundle...
ℹ ｢hot｣: webpack: Compiling...
 98% after emittingCompleted client bundle in 2.881s!
ℹ ｢hot｣: webpack: Compiling Done
ℹ ｢wdm｣: Version: webpack 4.11.0
Time: 13881ms
Built at: 05.06.2018 16:44:22
Entrypoint main = main.css index.js main.0a655b4a95cd904b6410.hot-update.js
ℹ ｢wdm｣: Compiled successfully.
```

**Comment by swernerx** [comment]
Content of the transferred files:

```json
{"h":"59574152234a740a3f28","c":{"main":true}}
```

```js
webpackHotUpdate("main",[])
````
This seems to be a little too light.

**Comment by swernerx** [comment]
BTW I am using the new dev-server "webpack-serve". So no SSR in place right now.

**Comment by swernerx** [comment]
There are no warnings. It's really just not working. Webpack output itself looks fine.

**Comment by zackljackson** [comment]
Okay, I think know the issue. Do you have a repo that I can fix it on? I think for your immediate relief, im going to add hot:true to the plugin part

**Comment by swernerx** [comment]
Sorry, it's in a private repo of a customer of mine. Can't really share that.

**Comment by swernerx** [comment]
I can hack into the code of your plugin though to try it out... if you give me an idea of where to look.

**Comment by zackljackson** [comment]
No I fully understand! Give me a second and ill release a @next on npm which should resolve your issue

**Comment by manjula91** [comment]
hmr : true should solve this issue

**Comment by zackljackson** [comment]
@manjula91 on the plugin or setting it in the webpack options?

**Comment by zackljackson** [comment]
Okay guys Im testing a fix for this @manjula91 @swernerx 

Try `npm I extract-css-chunks-webpack-plugin@3.0.4-beta.1`

How I currently expect things.

If you can send a snippet of were you define HOT, it will be very helpful!
Alternatively please tag me in anything you plan to release open source so I can automate it as well 

```js
  devServer: {
    contentBase: __dirname,
    hot: true,
  },
```

**OR**

```js
    entry: {
        dev: './dev/javascript',
        index: [
            'webpack/hot/only-dev-server',
            `webpack-hot-middleware/client?path=${config.host}:${config.port}/__webpack_hmr`,
            './src/client/index.js'
        ]
    },
```

**Comment by zackljackson** [comment]
Or `npm I extract-css-chunks-webpack-plugin@next`

**Comment by zackljackson** [comment]
@swernerx 

Let me know fi this helps, im at work otherwise id dedicate full time to fix it. if you still have issues.


What I had was a conditional checking the the process.env.NODE_ENV === development

secondly I have a function that determines is you use HMR 

However id either are false. it won't add. For the time being you can just add hot:true to the plugin
Ill leave that there but will also look at your specific plugin at check if I can automate the test as well

Might appreciate the patience

**Comment by swernerx** [comment]
Thanks. The new flag is fixing my issue.

**Comment by hauckwill** [comment]
Still not working for me with `hot: true`, sadly. A hot update is triggered but it's empty:

```
webpackHotUpdate("main",[])
```

Using `^3.0.4-beta.1`.

**Comment by ScriptedAlchemy** [comment]
Uninstall reinstall @latest it’s out of beta.

**Comment by hauckwill** [comment]
No luck on @latest unfortunately.

**Comment by ScriptedAlchemy** [comment]
@hauckwill if you can, throw me a repo and ill check it

**Comment by hauckwill** [comment]
Unfortunately the full repo is not mine to share, but here's my webpack config:
https://gist.github.com/hauckwill/e3fe0fe3c508bbb0e8a32c17707262c1

Using `"extract-css-chunks-webpack-plugin": "^3.0.5"`

Really appreciate the help and the work you've put into this.

**Comment by ScriptedAlchemy** [comment]
@hauckwill Ping me on https://gitter.im/Reactlandia

It will be easier & faster for me to assist over a private IM
